### PR TITLE
overlay.nix: spring-cleaning

### LINKED ISSUE #407: Review attributes in overlay.nix
Springtime cleaning!

Are there any packages we don't need anymore? If so, removing them would reduce closure size and build times, and improve maintainability.

Some possible candidates:

- [ ] dappsys
- [ ] dappsys-merged
- [ ] dapp-tests (cfr. #404)
- [ ] known-contracts
- [ ] dapp-which
- [ ] dapp2
- [ ] evmdis
- [x] setzer/terra/chief (cfr. #377)
- [ ] myetherwallet
- [ ] dafny
- [ ] qrtx/qrtx-term

### PR DESCRIPTION
**Pull request by xwvvvvwx**
This cleans up a bunch of old attributes from `overlay.nix` that are (as far as I'm aware) currently unused, this removes everything from https://github.com/dapphub/dapptools/issues/407 and a few more.

The removed attributes are:

- `profilingHaskellPackages`
- `callSolidityPackage`
- `dapps`
- `evmdis`
- `known-contracts`
- `dapp-which`
- `celf`
- `myEtherWallet`
- `dafny`

If you use something on this list, I'm super happy to keep it, I just prefered to be aggresive and then readd anything that was actually useful if needed.

Closes #407.

### CODE REVIEW

File: overlay.nix [hunk 1]
```diff
@@ -130,62 +105,7 @@ in rec {
     '';
   });
 
-  qrtx = self.bashScript {
```
Comment (asymmetric) [inline overlay.nix:133]: Is this used by anybody? Could be cool in theory

Comment (MrChico) [reply overlay.nix:133]: what is it?

Comment (xwvvvvwx) [reply overlay.nix:133]: I think it prints a tx as a qr code? I guess it's useful if you're doing offline signing.

**Comment by asymmetric** [comment]
[This](

https://github.com/dapphub/dapptools/tree/c351dde206dc8d6f3237118f251cffb18553f292/nix/build-dapp.nix#L43) should've probably been removed in #438 - can you remove it here instead?

File: overlay.nix (hunk 1 above)
Comment (rainbreak) [reply overlay.nix:133]: Yes, it's for using air-gapped signing devices. I vote leave these in.

File: nix/evmdis.nix [hunk 2]
```diff
@@ -1,21 +0,0 @@
-{ stdenv, lib, buildGoPackage, fetchFromGitHub }:
```
Comment (rainbreak) [inline nix/evmdis.nix:1]: I haven't used evmdis for a while, but we do advertise it on dapp.tools

File: overlay.nix [hunk 3]
```diff
@@ -108,8 +85,6 @@ in rec {
 
   evmdis = self.callPackage ./nix/evmdis.nix {};
 
-  token = self.callPackage (import ./src/token) {};
```
Comment (rainbreak) [inline overlay.nix:111]: I think this may have some users still, not sure.

Comment (rainbreak) [reply overlay.nix:111]: Could just delete and see if there are complaints.

**Comment by MrChico** [comment]
`dafny`, `celf` and `evmdis` all have equal version or newer in nixpkgs. So I'm fine with removing them. Does Mr. @livnev use `token` maybe? There's nothing I would really miss here.

File: overlay.nix (hunk 3 above)
Comment (livnev) [reply overlay.nix:111]: We should keep this.

File: overlay.nix (hunk 1 above)
Comment (livnev) [reply overlay.nix:133]: This is cool, +1 for keep

File: nix/evmdis.nix (hunk 2 above)
Comment (xwvvvvwx) [reply nix/evmdis.nix:1]: The version w

Comment (xwvvvvwx) [reply nix/evmdis.nix:1]: The version we have here is more than 3 years old, I had assumed that it would be broken on most new contracts?

**Comment by xwvvvvwx** [comment]
I put `qrtx`, `qrtx-term`, and `token` back.

**Comment by MrChico** [comment]
> [This](https://github.com/dapphub/dapptools/tree/c351dde206dc8d6f3237118f251cffb18553f292/nix/build-dapp.nix#L43) should've probably been removed in #438 - can you remove it here instead? Or too OT?

plzz

File: nix/evmdis.nix (hunk 2 above)
Comment (xwvvvvwx) [reply nix/evmdis.nix:1]: There is a maintained and up to date version in `nixpkgs` fwiw: https://github.com/NixOS/nixpkgs/blob/84cf00f98031e93f389f1eb93c4a7374a33cc0a9/pkgs/development/tools/analysis/evmdis/default.nix

**Comment by xwvvvvwx** [comment]
I'm not sure what tag to give the release after this has merged?

**Comment by asymmetric** [comment]
Do we have to make a release for this? It's mostly internal (and deprecation of stuff nobody used, hopefully), so maybe it should just be bundled with the next user-facing release.

**Comment by xwvvvvwx** [comment]
I'm fine with that.

**Pull request merged**